cookbook/workflows/_06_advanced_concepts/_06_other/workflow_history/enable_history_for_step.py

```diff
@@ -0,0 +1,108 @@
+"""
+This example shows step-level add_workflow_history control.
+Only the Content Creator step gets workflow history to avoid repeating previous content.
+
+Workflow: Research → Content Creation (with history) → Publishing
+"""
+
+from agno.agent import Agent
+from agno.db.sqlite import SqliteDb
+from agno.models.openai import OpenAIChat
+from agno.workflow.step import Step
+from agno.workflow.workflow import Workflow
+
+
+def create_content_creation_workflow():
+    """Content creation workflow where only the content creator uses workflow history"""
+
+    # Research agent - doesn't need workflow history, just does research
+    research_agent = Agent(
```

Comment:
I'd say this one also benefits, because if you say something like "Now make a twitter thread about the same topic" after the previous request, it would know to do the research correctly.